docs/rules/prefer-object-from-entries.md

```diff
@@ -0,0 +1,80 @@
+# Prefer using `Object.fromEntries(…)` to transform a list of key-value pairs into an object
+
+When transforming a list of key-value pairs into an object, [`Object.fromEntries(…)`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/fromEntries) should be preferred.
+
+This rule is fixable for simple cases.
+
+## Fail
+
+```js
+const object = pairs.reduce(
+	(object, [key, value]) => ({...object, [key]: value}),
+	{}
+);
+```
+
+```js
+const object = pairs.reduce(
+	(object, [key, value]) => ({...object, [key]: value}),
+	Object.create(null)
+);
+```
+
+```js
+const object = pairs.reduce(
+	(object, [key, value]) => Object.assign(object, {[key]: value}),
+	{}
+);
+```
+
+```js
+const object = pairs.reduce(addPairToObject, {});
+```
+
+```js
+const object = _.fromPairs(pairs);
+```
+
+## Pass
+
+```js
+const object = Object.fromEntries(pairs);
+```
+
+```js
+const object = new Map(pairs);
+```
+
+## Options
+
+Type: `object`
+
+### functions
+
+Type: `string[]`
+
+You can also check custom functions that transforms pairs.
+
+`lodash.fromPairs()` and `_.fromPairs()` are checked by default.
```

Comment:
It's not clear from the docs here whether setting this option overrides the default checked functions or extends it. That should be made more clear.

Reply:
How about 

> `lodash.fromPairs()` and `_.fromPairs()` are always checked.


Note to myself, [`prefer-array-flat` use the same](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-array-flat.md#functions).

Reply:
👍